bindsnet/network/nodes.py

```diff
@@ -392,12 +407,21 @@ def forward(self, x: torch.Tensor) -> None:
         self.refrac_count[self.refrac_count != 0] -= self.dt
 
         # Check for spiking neurons.
-        self.s = self.v >= self.thresh
+        if self.noise_std > 0:
+            noise = torch.randn(self.s.shape)*self.noise_std
+            self.s = self.v + noise >= self.thresh
+        else:
+            self.s = self.v >= self.thresh
+
 
         # Refractoriness and voltage reset.
         self.refrac_count.masked_fill_(self.s, self.refrac)
         self.v.masked_fill_(self.s, self.reset)
 
+        # voltage clipping to lowerbound
+        self.v.masked_fill_(self.v < self.lbound, self.lbound)
```

Comment:
This fails when `self.lbound` is `tensor-valued`. Can you fix this?

E.g.,

```
v = torch.randn(100)
lbound = torch.randn(100)
v.masked_fill_(v < lbound, lbound)

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-8-e8697cbeaaab> in <module>
----> 1 v.masked_fill_(v < lbound, lbound)

RuntimeError: masked_fill_ only supports a 0-dimensional value tensor, but got tensor with 1 dimension(s).
```

Reply:
Actually I supposed lbound is a single scalar value, which means that the whole layer shares the same lbound value. But if you still think supporting neuron-specific lbound is needed, I'll try that.

Reply:
Got it

Reply:
I've just updated